### ISSUE TITLE
ESP8266 returning wrong hour en minutes values

### ISSUE THREAD

**Issue by MrSnoebel** [arduino-libraries/NTPClient#61]
I'm using the library for setting the time to be used on a clock. 
This is a real clock for in the living room, problem happens on different clocks on different locations exactly at the same time ! The value returned is always 19 for the hours() and 29 for the minutes(). It does this about once a day at a random time, but mostly in the evenings. So it must be code related.

Details for the bits that matter for this problem, very straightforward I thought..

#include <NTPClient.h>
#include <WiFiUdp.h>

WiFiUDP ntpUDP;
//First number is timeoffset (1 uur), second is updateinterval (4 uur)
NTPClient timeClient(ntpUDP, "nl.pool.ntp.org", 3600, 14400);

void setup(){ 
timeClient.begin(); 
}

Void loop(){
timeClient.update();
}

int minuten = timeClient.getMinutes();
int uur = timeClient.getHours();

Any help in any direction is very appreciated !!

**Comment by jbrown123** [comment]
I'm doing something very similar on an ESP8266 and it works fine. What error are you seeing?  How far off is the time?

A couple of things to check. First, realize that update() returns true/false to indicate if the time is valid / didn't update.  It doesn't appear that you are checking this value and thus your time may not be set because of some issue (no net connection, invalid NTP server, etc.).

Second, realize that the time is in GMT, or in your case GMT + 3600 seconds (1 hour). Is that what you are expecting?

**Comment by MrSnoebel** [comment]
Thx for responding !

The time is all over the place, here is an example of what was happening : 

Formatted time : 10:06:11 uren : 10 minuten : 6 Epoch : 1546682771 millis : 12547
Formatted time : 10:06:21 uren : 10 minuten : 6 Epoch : 1546682781 millis : 22547
Formatted time : 10:06:31 uren : 10 minuten : 6 Epoch : 1546682791 millis : 32547
Formatted time : 10:06:41 uren : 10 minuten : 6 Epoch : 1546682801 millis : 42547
Formatted time : 10:06:51 uren : 10 minuten : 6 Epoch : 1546682811 millis : 52547
Formatted time : 10:06:02 uren : 10 minuten : 6 Epoch : 1546682762 millis : 62548
Formatted time : 10:06:12 uren : 10 minuten : 6 Epoch : 1546682772 millis : 72548
Formatted time : 10:06:22 uren : 10 minuten : 6 Epoch : 1546682782 millis : 82548
Formatted time : 10:06:32 uren : 10 minuten : 6 Epoch : 1546682792 millis : 92548
Formatted time : 10:06:42 uren : 10 minuten : 6 Epoch : 1546682802 millis : 102548
Formatted time : 10:06:52 uren : 10 minuten : 6 Epoch : 1546682812 millis : 112548
Formatted time : 10:06:03 uren : 10 minuten : 6 Epoch : 1546682763 millis : 122548
Formatted time : 10:06:13 uren : 10 minuten : 6 Epoch : 1546682773 millis : 132548
Formatted time : 10:06:23 uren : 10 minuten : 6 Epoch : 1546682783 millis : 142548
Formatted time : 10:06:33 uren : 10 minuten : 6 Epoch : 1546682793 millis : 152548
Formatted time : 10:06:43 uren : 10 minuten : 6 Epoch : 1546682803 millis : 162548
Formatted time : 10:06:53 uren : 10 minuten : 6 Epoch : 1546682813 millis : 172548
Formatted time : 10:06:04 uren : 10 minuten : 6 Epoch : 1546682764 millis : 182548
Formatted time : 10:06:14 uren : 10 minuten : 6 Epoch : 1546682774 millis : 192548
Formatted time : 10:06:24 uren : 10 minuten : 6 Epoch : 1546682784 millis : 202548
Formatted time : 10:06:34 uren : 10 minuten : 6 Epoch : 1546682794 millis : 212548
Formatted time : 10:06:44 uren : 10 minuten : 6 Epoch : 1546682804 millis : 222548
Formatted time : 10:06:54 uren : 10 minuten : 6 Epoch : 1546682814 millis : 232548
Formatted time : 10:06:05 uren : 10 minuten : 6 Epoch : 1546682765 millis : 242548
Formatted time : 10:06:15 uren : 10 minuten : 6 Epoch : 1546682775 millis : 252548
Formatted time : 10:06:25 uren : 10 minuten : 6 Epoch : 1546682785 millis : 262548
Formatted time : 10:06:35 uren : 10 minuten : 6 Epoch : 1546682795 millis : 272548
Formatted time : 10:06:45 uren : 10 minuten : 6 Epoch : 1546682805 millis : 282548
Formatted time : 10:06:55 uren : 10 minuten : 6 Epoch : 1546682815 millis : 292548
Formatted time : 10:06:06 uren : 10 minuten : 6 Epoch : 1546682766 millis : 302548
Formatted time : 10:06:16 uren : 10 minuten : 6 Epoch : 1546682776 millis : 312548
Formatted time : 10:06:26 uren : 10 minuten : 6 Epoch : 1546682786 millis : 322548

Beside this it also jumps to 07:28 for minutes, this was for me the main issue and the reason why it was seen at all.

Sorry for the dutch, uren = hours, minuten = minutes. I use only those two for the clock.

I couldn't get it working, so I ditched this library and now use the time.h from the core ESP library (2.4.2).
That seems rock solid and also is updated by NTP.

**Comment by jbrown123** [comment]
Update interval is in ms not in seconds.  See lines 95-102, particularly line 96 (note use of millis())
https://github.com/arduino-libraries/NTPClient/blob/020aaf868d53331a6f2909f7f6a475998c527ae4/NTPClient.cpp#L95-L102

So your setting of 14400 in the original code you posted is actually 14.4 seconds, not 4 hours.

The code is behaving as if it has the original 60 second update interval set.  I note from your latest post that it basically resets the time every 60 seconds / 60000 millis.  I wonder if your time server simply wont respond any more often.

Are you sure you are using the latest version of all the libraries?

I'd be interested in seeing the actual code that generated the above output. What you posted before is obviously not that code.  Can you create a very simple program that demonstrates the problem and then post both the code and the output (similar to the output you sent previously)?

Just to confirm, you are using an ESP8266, right?

**Comment by MrSnoebel** [comment]
Nice catch for the update time, indeed I missed the fact that it's ms.

I was using the following versions : 
esp8266 by ESP8266 community version 2.4.2
WiFiManager by tzapu version 0.14.0
FastLED by Daniel Garcia version 3.2.1
NTPClient by Fabrice Weinberg version 3.1.0 (but tested also 3.0.0, which did not work at all)

Yes, I'm using a NodeMCU v3 with a ESP8266 (ESP-12E).

To test I did :  
  EVERY_N_MILLISECONDS(10000){
  Serial.print("Formatted time : ");Serial.print(timeClient.getFormattedTime());
  Serial.print(" uren : ");Serial.print(timeClient.getHours());
  Serial.print(" minuten : ");Serial.print(timeClient.getMinutes());
  Serial.print(" Epoch : ");Serial.print(timeClient.getEpochTime());
  Serial.print(" millis : ");Serial.println(millis());
  };

But I switched to the build in time.h with the core, all is fine now for me.
